middle_end/flambda/naming/name_occurrences.ml

```diff
@@ -524,6 +524,9 @@ end)
 module For_code_ids = For_one_variety_of_names (struct
   include Code_id
   (* We never bind [Code_id]s using [Name_abstraction]. *)
+  (* CR gbury: should we nevertheless call Name_permutation.apply_code_id,
+     at least for theoretical correctness ? (and if not, these should
+     probably be fatal_errors) *)
```

Comment:
You should definitively call `Name_permutation.apply_code_id`. The fact that we never bind code IDs using `Name_abstraction` is irrelevant now that we allow renaming code IDs on the compilation unit scope. I believe that nothing relies on the name occurrences of code IDs after simplification, so currently this is harmless, but it should be fixed anyway.

Reply:
Fixed